### ISSUE TITLE
Copying failed during building since there is no generatedSchema.usda

### ISSUE THREAD

**Issue by hongloull** [AnimalLogic/AL_USDMaya#13]
Hi There, Got below error during building it:

-- Boost version: 1.55.0
-- Found the following Boost libraries:
--   python
-- Building AL_USDMayaSchemas
CMake Warning (dev) at schemas/AL/usd/schemas/CMakeLists.txt:60 (get_target_property):
  Policy CMP0026 is not set: Disallow use of the LOCATION target property.
  Run "cmake --help-policy CMP0026" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The LOCATION property should not be read from target "AL_USDMayaSchemas".
  Use the target name directly with add_custom_command, or use the generator
  expression $<TARGET_FILE>, as appropriate.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at schemas/AL/usd/schemas/CMakeLists.txt:75 (file):
  file COPY cannot find
  "/home/mjun/workspace/AL_USDMaya/build/schemas/AL/usd/schemas/generatedSchema.usda".

Any ideas about this? Thanks a lot.

**Comment by robthebloke** [comment]
Noted. It's way beyond my bedtime right now, but I have a feeling as to what the cause may be. Whilst usdmaya should not depend on the schemas to build, it may do for testing purposes only. (Actually, that's a lie, some metadata registration is a requirement for certain features to work - but that should only show up in failed tests - and definitely not the builds). I think we have some work our end to separate some of those concerns - we actually noticed an issue yesterday that sounds very much related.

We'll take a look in the morning, and get back to you. i suspect (but am not going to promise) that we can find a workaround fairly quickly (if my suspicions are correct), otherwise we'll bump that to top priority, and do our best!

**Comment by fabal** [comment]
Hi @hongloull ,
could you send us the full cmake log? It looks like you're having a schema generation issue and the error you're getting could be a consequence of that.

**Comment by iceprincefounder** [comment]
Hi @fabal ,
I got a same problem to generate project,I have sent you my cmake log,thanks.

[CMakeError.txt](https://github.com/AnimalLogic/AL_USDMaya/files/1213590/CMakeError.txt)
[CMakeOutput.txt](https://github.com/AnimalLogic/AL_USDMaya/files/1213588/CMakeOutput.txt)

KT

**Comment by fabal** [comment]
Thanks. Could you send us the log of the cmake _configuration_ step? (the result of the `cmake` command, before you call `make`).
For now you should be able to bypass this error by adding these two flags to your `cmake` command line: `-DBUILD_USDMAYA_SCHEMAS=0 -DBUILD_USDMAYA_TRANSLATORS=0`.
Also the logs you've sent suggests `gcc` is called instead of `g++`. Make sure you've got g++ properly configured.

**Comment by JasonGilholme** [comment]
I'm experiencing the same thing.  Initially my pxrConfig.cmake file was trying to call a file that didn't exist, I've patched that and still get this error during configuration.

```
-- Building AL_USDMayaSchemas
CMake Warning (dev) at schemas/AL/usd/schemas/CMakeLists.txt:60 (get_target_property):
  Policy CMP0026 is not set: Disallow use of the LOCATION target property.
  Run "cmake --help-policy CMP0026" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The LOCATION property should not be read from target "AL_USDMayaSchemas".
  Use the target name directly with add_custom_command, or use the generator
  expression $<TARGET_FILE>, as appropriate.

This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at schemas/AL/usd/schemas/CMakeLists.txt:75 (file):
  file COPY cannot find
  "/opt/build/dev/AL_USDMaya/_rez/build/platform-linux/arch-x86_64/os-CentOS-6/app_maya-2017.1/AL_USDMaya/src/AL_USDMaya-build/schemas/AL/usd/schemas/generatedSchema.usda".


-- Building AL_USDMayaTranslators
CMake Warning (dev) at translators/CMakeLists.txt:20 (get_target_property):
  Policy CMP0026 is not set: Disallow use of the LOCATION target property.
  Run "cmake --help-policy CMP0026" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  The LOCATION property should not be read from target
  "AL_USDMayaTranslators".  Use the target name directly with
  add_custom_command, or use the generator expression $<TARGET_FILE>, as
  appropriate.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
```

**Comment by JasonGilholme** [comment]
Managed to get a build out after adding some extra find_package() calls and using these lines to avoid the schema and file translator builds:

-DBUILD_USDMAYA_SCHEMAS=OFF
-DBUILD_USDMAYA_TRANSLATORS=OFF

translators/MayaReference.cpp is trying to include some files from AL/usdmaya and AL/usd which don't exist in the repo.

**Comment by fabal** [comment]
Good to know that you've been able to build it :thumbsup:
We've added these schemas/translators lately. The schema code is generated during the build and you've very likely hit this [issue](https://github.com/PixarAnimationStudios/USD/issues/158). We'll come up with a better solution very soon.

Feel free to create other separate issues for the other errors you're having.

**Comment by hongloull** [comment]
Thanks everyone!

**Comment by fabal** [comment]
Fixed in d165134158224136a545586f1b477aff0ba1d928
